BWC test failures for health/10_basic/cluster health basic test

### CI Link

https://gradle-enterprise.elastic.co/s/j2zhn5js47wcy

### Repro line

./gradlew ':qa:mixed-cluster:v8.5.3#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=health/10_basic/cluster health basic test}" -Dtests.seed=72A1B4518E0C3ECD -Dtests.bwc=true -Dtests.locale=es-UY -Dtests.timezone=America/Blanc-Sablon -Druntime.java=21

### Does it reproduce?

No

### Applicable branches

8.13, main

### Failure history

_No response_

### Failure excerpt

This is likely due to https://github.com/elastic/elasticsearch/pull/105789

```
java.lang.AssertionError: Failure at [health/10_basic:7]: expected [2xx] status code but api [health_report] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"invalid_index_name_exception","reason":"Invalid index name [_health_report], must not start with '_'.","index_uuid":"_na_","index":"_health_report","stack_trace":"[_health_report] org.elasticsearch.indices.InvalidIndexNameException: Invalid index name [_health_report], must not start with '_'.\n\tat org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.validateAliasOrIndex(IndexNameExpressionResolver.java:1266)\n\tat org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.innerResolve(IndexNameExpressionResolver.java:1202)\n\tat org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.resolve(IndexNameExpressionResolver.java:1145)\n\tat org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.resolveExpressions(IndexNameExpressionResolver.java:251)\n\tat org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:344)\n\tat org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndexNames(IndexNameExpressionResolver.java:314)\n\tat org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndexNames(IndexNameExpressionResolver.java:86)\n\tat org.elasticsearch.action.support.master.info.TransportClusterInfoAction.checkBlock(TransportClusterInfoAction.java:53)\n\tat org.elasticsearch.action.support.master.info.TransportClusterInfoAction.checkBlock(TransportClusterInfoAction.java:24)\n\tat org.elasticsearch.action.support.master.TransportMasterNodeAction.checkBlockIfStateRecovered(TransportMasterNodeAction.java:136)\n\tat org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.doStart(TransportMasterNodeAction.java:199)\n\tat org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:172)\n\tat org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:54)\n\tat org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:86)\n\tat org.elasticsearch.action.support.ActionFilter$Simple.apply(ActionFilter.java:53)\n\tat org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:84)\n\tat org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:61)\n\tat org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:71)\n\tat org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:67)\n\tat org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:71)\n\tat org.elasticsearch.transport.InboundHandler.handleRequest(InboundHandler.java:278)\n\tat org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:116)\n\tat org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:95)\n\tat org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:808)\n\tat org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:149)\n\tat org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:121)\n\tat org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:86)\n\tat org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.channelRead(Netty4MessageInboundHandler.java:63)\n\tat 
```

@nielsbauman I might have copied from the wrong one. Each bwc build is its own build scan and 5 different versions all failed. Let me enumerate the ones that failed:

 - 8.5.3: https://gradle-enterprise.elastic.co/s/v7gnviihwdkhc
 - 8.6.0: https://gradle-enterprise.elastic.co/s/4rvhmtsfkgg3k
 - 8.6.1: https://gradle-enterprise.elastic.co/s/j2zhn5js47wcy
 - 8.6.2: https://gradle-enterprise.elastic.co/s/27tcj6kbiru4a

Here is a full trace that is easier to read than the above one I pasted (I replaced the \n and \t):

```
  2> java.lang.AssertionError: Failure at [health/10_basic:7]: expected [2xx] status code but api [health_report] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"invalid_index_name_exception","reason":"Invalid index name [_health_report], must not start with '_'.","index_uuid":"_na_","index":"_health_report","stack_trace":"[_health_report] org.elasticsearch.indices.InvalidIndexNameException: Invalid index name [_health_report], must not start with '_'.
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.validateAliasOrIndex(IndexNameExpressionResolver.java:1261)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.innerResolve(IndexNameExpressionResolver.java:1197)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.resolve(IndexNameExpressionResolver.java:1141)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.resolveExpressions(IndexNameExpressionResolver.java:251)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:339)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndexNames(IndexNameExpressionResolver.java:314)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndexNames(IndexNameExpressionResolver.java:86)
                    at org.elasticsearch.action.support.master.info.TransportClusterInfoAction.checkBlock(TransportClusterInfoAction.java:53)
                    at org.elasticsearch.action.support.master.info.TransportClusterInfoAction.checkBlock(TransportClusterInfoAction.java:24)
                    at org.elasticsearch.action.support.master.TransportMasterNodeAction.checkBlockIfStateRecovered(TransportMasterNodeAction.java:135)
                    at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.doStart(TransportMasterNodeAction.java:199)
                    at org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:171)
                    at org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:53)
                    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:86)
                    at org.elasticsearch.action.support.ActionFilter$Simple.apply(ActionFilter.java:53)
                    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:84)
                    at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:61)
                    at org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:71)
                    at org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:67)
                    at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:71)
                    at org.elasticsearch.transport.InboundHandler.handleRequest(InboundHandler.java:278)
                    at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:116)
                    at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:95)
                    at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:808)
                    at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:149)
                    at org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:121)
                    at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:86)
                    at org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.channelRead(Netty4MessageInboundHandler.java:63)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
                    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
                    at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
                    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
                    at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
                    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
                    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
                    at io.netty.channel.AbstractChannelHandlerContext.inv…	
        at __randomizedtesting.SeedInfo.seed([61D26802C167C7CF:E98657D86F9BAA37]:0)	
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:561)	
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:504)	
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)	
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)	
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)	
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)	
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)	
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)	
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)	
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)	
        at java.base/java.lang.Thread.run(Thread.java:1583)	
        Caused by:	
        java.lang.AssertionError: expected [2xx] status code but api [health_report] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"invalid_index_name_exception","reason":"Invalid index name [_health_report], must not start with '_'.","index_uuid":"_na_","index":"_health_report","stack_trace":"[_health_report] org.elasticsearch.indices.InvalidIndexNameException: Invalid index name [_health_report], must not start with '_'.
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.validateAliasOrIndex(IndexNameExpressionResolver.java:1261)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.innerResolve(IndexNameExpressionResolver.java:1197)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.resolve(IndexNameExpressionResolver.java:1141)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.resolveExpressions(IndexNameExpressionResolver.java:251)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:339)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndexNames(IndexNameExpressionResolver.java:314)
                    at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndexNames(IndexNameExpressionResolver.java:86)
                    at org.elasticsearch.action.support.master.info.TransportClusterInfoAction.checkBlock(TransportClusterInfoAction.java:53)
                    at org.elasticsearch.action.support.master.info.TransportClusterInfoAction.checkBlock(TransportClusterInfoAction.java:24)
                    at org.elasticsearch.action.support.master.TransportMasterNodeAction.checkBlockIfStateRecovered(TransportMasterNodeAction.java:135)
                    at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.doStart(TransportMasterNodeAction.java:199)
                    at org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:171)
                    at org.elasticsearch.action.support.master.TransportMasterNodeAction.doExecute(TransportMasterNodeAction.java:53)
                    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:86)
                    at org.elasticsearch.action.support.ActionFilter$Simple.apply(ActionFilter.java:53)
                    at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:84)
                    at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:61)
                    at org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:71)
                    at org.elasticsearch.action.support.HandledTransportAction$TransportHandler.messageReceived(HandledTransportAction.java:67)
                    at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:71)
                    at org.elasticsearch.transport.InboundHandler.handleRequest(InboundHandler.java:278)
                    at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:116)
                    at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:95)
                    at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:808)
                    at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:149)
                    at org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:121)
                    at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:86)
                    at org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.channelRead(Netty4MessageInboundHandler.java:63)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
                    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
                    at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
                    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
                    at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
                    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
                    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
                    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChanne…	
            at org.junit.Assert.fail(Assert.java:89)	
            at org.elasticsearch.test.rest.yaml.section.DoSection.checkResponseException(DoSection.java:529)	
            at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:385)	
            at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:541)	
            ... 40 more
```

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/health/10_basic.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/health/10_basic.yml
@@ -2,7 +2,7 @@
 "cluster health basic test":
   - skip:
       version: "- 8.6.99"
-      reason: "health was only added in 8.2.0, and master_is_stable in 8.4.0"
+      reason: "health was added in 8.2.0, master_is_stable in 8.4.0, and REST API updated in 8.7"
 
   - do:
       health_report: { }

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/health/10_basic.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/health/10_basic.yml
@@ -1,7 +1,7 @@
 ---
 "cluster health basic test":
   - skip:
-      version: "- 8.3.99"
+      version: "- 8.6.99"
       reason: "health was only added in 8.2.0, and master_is_stable in 8.4.0"
 
   - do:

